contracts/blender/src/contract.rs

```diff
@@ -160,6 +176,31 @@ mod blender {
             self.current_root()
         }
 
+        /// Retrieve the path from the leaf to the root. `None` if the leaf does not exist.
+        #[ink(message, selector = 4)]
+        pub fn merkle_path(&self, leaf_idx: u32) -> Option<MerklePath> {
+            if self.max_leaves > leaf_idx || leaf_idx >= self.next_free_leaf {
+                return None;
+            }
+
+            let mut current_idx = leaf_idx;
+            let mut path: MerklePath = vec![];
+            while current_idx > 1 {
+                let (left_sibling_idx, right_sibling_idx) = Self::get_sibling_indices(current_idx);
+                path.push(self.tree_value(left_sibling_idx));
+                path.push(self.tree_value(right_sibling_idx));
+                current_idx /= 2;
```

Comment:
You don't need to add both to the path, just the one which does not equal `current_idx`. Given the leaf, and the ones I propose, you can reconstruct the path.

Reply:
optimized